.github/workflows/circuit_check.yml

```diff
@@ -0,0 +1,239 @@
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ~/.risc0
+          key: ${{ runner.os }}-risc0-toolchain-${{ env.RISC0_TOOLCHAIN_VERSION }}
+
+      - name: Install risc0 prereqs (only when cache cold)
+        if: steps.cache-toolchain.outputs.cache-hit != 'true'
+        uses: ./.github/actions/test-prerequisites
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install r0vm (only when cache cold)
+        if: steps.cache-toolchain.outputs.cache-hit != 'true'
+        run: rzup install r0vm
+
+  bridge_circuit_check:
+    name: Bridge Circuit Method ID Check
+    runs-on: ubicloud-standard-8
+    needs: toolchain
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Cache Cargo & RISC Zero toolchain
+        id: cache-toolchain
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ~/.risc0
+          key: ${{ runner.os }}-risc0-toolchain-${{ env.RISC0_TOOLCHAIN_VERSION }}
+
+      - name: Install risc0 prereqs (only if cache missed)
+        if: steps.cache-toolchain.outputs.cache-hit != 'true'
+        uses: ./.github/actions/test-prerequisites
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install r0vm (only if cache missed)
+        if: steps.cache-toolchain.outputs.cache-hit != 'true'
+        run: rzup install r0vm
+
+      - name: Ensure toolchain is in PATH
+        run: |
+          echo "$HOME/.cargo/bin" >> "$GITHUB_PATH"
+          echo "$HOME/.risc0/bin" >> "$GITHUB_PATH"
+
+      - name: Compute bridge circuit pre-build method ID
+        id: pre_id
+        working-directory: risc0-circuits
+        run: |
+          PRE_ID=$(r0vm --elf elfs/mainnet-bridge-circuit-guest.bin --id | tr -d '[:space:]')
+          echo "pre_id=$PRE_ID" >> "$GITHUB_OUTPUT"
+
+      - name: Build Bridge Circuit
+        working-directory: risc0-circuits/bridge-circuit
+        run: cargo build -p bridge-circuit --release
+
+      - name: Compute bridge circuit post-build method ID
+        id: post_id
+        working-directory: risc0-circuits
+        run: |
+          POST_ID=$(r0vm --elf elfs/mainnet-bridge-circuit-guest.bin --id | tr -d '[:space:]')
+          echo "post_id=$POST_ID" >> "$GITHUB_OUTPUT"
+
+      - name: Verify bridge circuit method ID unchanged
+        run: |
+          if [ "${{ steps.pre_id.outputs.pre_id }}" != "${{ steps.post_id.outputs.post_id }}" ]; then
+            echo "Bridge method ID has changed from ${{ steps.pre_id.outputs.pre_id }} to ${{ steps.post_id.outputs.post_id }}"
+            exit 1
+          else
+            echo "Bridge method ID is unchanged: ${{ steps.pre_id.outputs.pre_id }}"
+          fi
+
+  header_chain_circuit_check:
```

Comment:
can we use a matrix to write a single job instead of two?

```
jobs:
  circuit_check:
    strategy:
      matrix:
        circuit_name: ["header-chain", "work-only"]
  ...
 ```